src/java.base/share/classes/sun/security/pkcs/PKCS9Attribute.java

```diff
@@ -600,49 +528,32 @@ public void encode(DerOutputStream out) {
             } // close scope
             break;
 
-        case 9:     // extended-certificate attribute -- not supported
-            throw new IllegalArgumentException("PKCS9 extended-certificate " +
-                                  "attribute not supported.");
-            // break unnecessary
-        case 10:    // issuerAndserialNumber attribute -- not supported
-            throw new IllegalArgumentException("PKCS9 IssuerAndSerialNumber " +
-                                  "attribute not supported.");
-            // break unnecessary
-        case 11:    // RSA DSI proprietary
-        case 12:    // RSA DSI proprietary
-            throw new IllegalArgumentException("PKCS9 RSA DSI attributes " +
-                                  "11 and 12, not supported.");
-            // break unnecessary
-        case 13:    // S/MIME unused attribute
-            throw new IllegalArgumentException("PKCS9 attribute #13 not supported.");
-            // break unnecessary
-
-        case 14:     // ExtensionRequest
+        case KnownOIDs.ExtensionRequest:
             {
                 DerOutputStream temp2 = new DerOutputStream();
                 CertificateExtensions exts = (CertificateExtensions)value;
                 exts.encode(temp2, true);
                 temp.write(DerValue.tag_Set, temp2.toByteArray());
             }
             break;
-        case 15:    // SMIMECapability
-            throw new IllegalArgumentException("PKCS9 attribute #15 not supported.");
-            // break unnecessary
 
-        case 16:    // SigningCertificate
+        case KnownOIDs.SigningCertificate:
             {
                 DerOutputStream temp2 = new DerOutputStream();
                 SigningCertificateInfo info = (SigningCertificateInfo)value;
                 temp2.writeBytes(info.toByteArray());
                 temp.write(DerValue.tag_Set, temp2.toByteArray());
             }
             break;
-        case 17:    // SignatureTimestampToken
-        case 18:    // CMSAlgorithmProtection
+        case KnownOIDs.SignatureTimestampToken:
+        case KnownOIDs.CMSAlgorithmProtection:
             temp.write(DerValue.tag_Set, (byte[])value);
             break;
 
-        default: // can't happen
+        default: //Should never happen
+            temp.writeBytes((byte[])value);
+            out.write(DerValue.tag_Sequence, temp.toByteArray());
+            return;
```

Comment:
For something that should never happen, we should throw exception to indicate the problem if it were to happen.